crates/sui-graphql-rpc/src/types/cursor.rs

```diff
@@ -234,6 +232,58 @@ impl Page<JsonCursor<usize>> {
     }
 }
 
+impl Page<JsonCursor<ConsistentIndexCursor>> {
+    /// Treat the cursors of this Page as indices into a range [0, total). Returns two booleans
+    /// indicating whether there is a previous or next page in the range, followed by an iterator of
+    /// cursors within that Page. Ignores whether each cursor's `c`, checkpoint sequence number, is
+    /// consistent, and returns an iterator of `JsonCursor<usize>`.
+    pub(crate) fn paginate_consistent_indices(
```

Comment:
nit: Note that in some places you use the `c` field of the consistent cursors returned by this call to populate the `checkpoint_viewed_at` field of the type being paginated, and in some cases you reuse whatever value was passed into `checkpoint_viewed_at` for the call to this function.

Both are correct, but for consistency, it would be good to standardise on one approach, because otherwise, a future refactor may create an issue.

Reply:
Thanks, yes we should standardize to using the cursor's checkpoint